cpp/unreal_plugins/SpEngine/Source/SpEngine/GameWorldService.h

```diff
@@ -178,8 +282,617 @@ class GameWorldService {
             });
 
         unreal_entry_point_binder->bindFuncUnreal("game_world_service", "get_class",
-            [this](const uint64_t& instance) -> uint64_t {
-                return reinterpret_cast<uint64_t>(reinterpret_cast<UObject*>(instance)->GetClass());
+            [this](const uint64_t& uobject) -> uint64_t {
+                return reinterpret_cast<uint64_t>(reinterpret_cast<UObject*>(uobject)->GetClass());
+            });
+
+        unreal_entry_point_binder->bindFuncUnreal("game_world_service", "get_static_class",
+            [this](const std::string& class_name) -> uint64_t {
+                return reinterpret_cast<uint64_t>(UnrealClassRegistrar::getStaticClass(class_name));
+            });
+
+        // TODO: How do we expose FActorSpawnParameters?
+        unreal_entry_point_binder->bindFuncUnreal("game_world_service", "spawn_actor",
+            [this](const std::string& class_name, const std::vector<double>& location, const std::vector<double>& rotation, const spawn_params) -> uint64_t {
+                SP_ASSERT(location.size() == 3);
+                SP_ASSERT(rotation.size() == 3);
+                return reinterpret_cast<uint64_t>(
+                    UnrealClassRegistrar::spawnActor(
+                        class_name,
+                        world_,
+                        FVector(location.at(0), location.at(1), location.at(2)),
+                        FRotator(rotation.at(0), rotation.at(1), rotation.at(2)),
+                        FActorSpawnParameters()));
```

Comment:
When you're ready, you can convert a string to an FSpActorSpawnParameters as follows:

```
FSpActorSpawnParameters sp_actor_spawn_parameters;
Unreal::setObjectPropertiesFromString(&sp_actor_spawn_parameters, spawn_params)
```

From here, I'm not sure what the cleanest way to convert between `FSpActorSpawnParameters` and `FActorSpawnParameters`, aside from the obvious ugly solution of copying every member. Doing a `memcpy` would work if we lay out all of the data in exactly the same order as the underlying Unreal struct, but this is brittle because the Unreal developers might change the data layout and we wouldn't notice.

Can we use inheritance to avoid the repetition? In other words, can we just make our `FSpActorSpawnParameters` inherit from `FActorSpawnParameters`? That would be ideal, because then we wouldn't need to explicitly duplicate the data layout.

Reply:
Update: no we can't use inheritance (or composition).